contract/src/main/kotlin/io/provenance/scope/loan/utility/FundingDataValidation.kt

```diff
@@ -0,0 +1,139 @@
+package io.provenance.scope.loan.utility
+
+import io.provenance.scope.util.toInstant
+import tech.figure.loan.v1beta1.Funding
+import tech.figure.util.v1beta1.ACH
+
+internal fun Funding?.isSet() = this !== null && (status.isSet() || started.isSet() || completed.isSet() || disbursementsCount > 0)
+
+internal val fundingValidation: ContractEnforcementContext.(Funding) -> Unit = { funding ->
+    if (funding.isSet()) {
+        funding.status.takeIf { it.isSet() }?.also { fundingStatus ->
+            requireThat(
+                (fundingStatus.status in listOf("UNFUNDED", "FUNDED", "INITIATED", "COMPLETED", "CANCELLED"))
```

Comment:
I think there is only these statuses:
 
- UNFUNDED - No funding record
- INITIATED 
- FUNDED
- CANCELLED

Reply:
👍🏽 I can removed `COMPLETED`. I had just included it since [the comment in the metadata-asset-model included it as a possible value for a disbursement status](https://github.com/provenance-io/metadata-asset-model/blob/62cf6ffa0acf18c94a96f2d2c7d25ded538b9430/src/main/proto/tech/figure/loan/v1beta1/loan.proto#L140), and we had seen that the disbursement status could be set to the same value as the funding status.